packages/devextreme/js/__internal/grids/grid_core/keyboard_navigation/scrollable_a11y.ts

```diff
@@ -0,0 +1,142 @@
+  }
+
+  protected rowsViewFocusHandler(event: any): void {
+    const $target = $(event.target);
+
+    this.translateFocusIfNeed(event, $target);
+
+    super.rowsViewFocusHandler(event);
+  }
+
+  private rowsViewFocusOutHandler(): void {
+    this.makeScrollableFocusableIfNeed();
+  }
+
+  private translateFocusIfNeed(event: any, $target: dxElementWrapper): void {
+    const needTranslateFocus = this.isScrollableNeedFocusable();
+    const isFirstCellFixed = this._isFixedColumn(0);
+
+    if (!needTranslateFocus || !isFirstCellFixed) {
+      return;
+    }
+
+    const $firstCell = this._rowsView.getCell({ rowIndex: 0, columnIndex: 0 });
+    const firstCellHasTabIndex = !!$firstCell.attr('tabindex');
+
+    // @ts-expect-error dxElementWrapper doesn't have overload for 'is' method
+    const notFixedCellIsTarget = $target.is(this._$firstNotFixedCell);
+
+    if (firstCellHasTabIndex && notFixedCellIsTarget) {
+      event.preventDefault();
+
+      this._focus($firstCell);
+    }
+  }
+
+  protected renderCompleted(e: any): void {
+    this._$firstNotFixedCell = this.getFirstNotFixedCell();
+    this.makeScrollableFocusableIfNeed();
+
+    super.renderCompleted(e);
+  }
+
+  protected _tabKeyHandler(eventArgs: any, isEditing: any): void {
+    const isCellPositionDefined = isDefined(this._focusedCellPosition)
+      && !isEmptyObject(this._focusedCellPosition);
+    const isOriginalHandlerRequired = !isCellPositionDefined
+      || (!eventArgs.shift && this._isLastValidCell(this._focusedCellPosition))
+      || (eventArgs.shift && this._isFirstValidCell(this._focusedCellPosition));
+
+    if (isOriginalHandlerRequired) {
+      if (this.isScrollableNeedFocusable()) {
+        this._$firstNotFixedCell?.removeAttr('tabIndex');
+      }
+    }
+
+    super._tabKeyHandler(eventArgs, isEditing);
+  }
+
+  private getFirstNotFixedCell(): dxElementWrapper | undefined {
+    const columns = this._columnsController.getVisibleColumns();
+
+    let columnIndex = -1;
+
+    for (let index = 0; index < columns.length; index += 1) {
+      const notFixedColumn = !columns[index].fixed;
+
+      if (notFixedColumn) {
+        columnIndex = index;
+        break;
+      }
+    }
+
+    if (columnIndex === -1) {
+      return undefined;
+    }
+
+    return this._rowsView._getCellElement(0, columnIndex) as dxElementWrapper | undefined;
+  }
+
+  private isScrollableNeedFocusable(): boolean {
+    const hasScrollable = !!this._rowsView.getScrollable();
+    const hasFixedTable = !!this._rowsView._fixedTableElement?.length;
+    const isCellsRendered = !!this._rowsView.getCellElements(0)?.length;
+
+    return hasScrollable && hasFixedTable && isCellsRendered;
+  }
+
+  private makeScrollableFocusableIfNeed(): void {
+    const needFocusable = this.isScrollableNeedFocusable();
+
+    if (!needFocusable) {
+      return;
+    }
+
+    if (this._$firstNotFixedCell) {
+      this._applyTabIndexToElement(this._$firstNotFixedCell);
+    }
```

Comment:
Let's union these two conditions into one:

```suggestion
    if (!needFocusable || !this._$firstNotFixedCell) {
      return;
    }

    this._applyTabIndexToElement(this._$firstNotFixedCell);
```